src/lib/core/CHIPPersistentStorageDelegate.h

```diff
@@ -126,6 +126,18 @@ class DLL_EXPORT PersistentStorageDelegate
      */
     virtual CHIP_ERROR SyncSetKeyValue(const char * key, const void * value, uint16_t size) { return CHIP_ERROR_NOT_IMPLEMENTED; }
 
+    /**
+     * @brief
+     *   This is a synchronous Set API.
+     *   This API should be used sparingly, since it may block for some duration.
+     *
+     *   Set the value for the key to a null terminated string.
+     *
+     * @param[in] key Key to be set
+     * @param[in] value Value to be set
+     */
+    virtual CHIP_ERROR SyncSetKeyValue(const char * key, const char * value) { return CHIP_ERROR_NOT_IMPLEMENTED; }
```

Comment:
Could we use and implement the void* value instead?

I find that we often need to serialize binary data: all the admin info and device info are binary and we force base64 encoding/decoding everywhere. It seems more convenient to have binary support only and interfaces for that already exist.

I think this proliferates more API split: we have sync/async, we have char * and void * which are likely not interoperable (null terminator handling and data format in general, it seems some implementation implement some and not the rest). This is the problem I was trying to solve over many PRs and it seems to get only worse. In this case we add a 'SyncSetKeyValue' on char * that is only implemented on Darwin and will not exist on many other platforms

Reply:
We need another PR to start storing thing as void * and len instead of char *. I effectively hotfixed this without adding too much functional logic to the existing code. Your points are valid though.

Reply:
Looking at this a bit more. The Darwin APIs are all (Char *) based. To do this using the void * API, the darwin delegate implmentation needs to be rewritten to use `NSData` or generic pointers `id` (probably NSData?). I'm currently not planning to do that in this PR. Which means this is kinda dead in the water until we change the darwin APIs first.